test/new-e2e/pkg/utils/e2e/client/vm_client.go

```diff
@@ -60,3 +70,25 @@ func (vmClient *vmClient) CopyFolder(srcFolder string, dstFolder string) {
 	err := clients.CopyFolder(vmClient.client, srcFolder, dstFolder)
 	require.NoError(vmClient.t, err)
 }
+
+func (vmClient *vmClient) setEnvVariables(command string, envVar executeparams.EnvVar) string {
+
+	cmd := ""
+	if vmClient.os.GetType() == commonos.WindowsType {
+		for envName, envValue := range envVar {
+			cmd += fmt.Sprintf("$env:%s='%s'; ", envName, envValue)
+		}
+		cmd += fmt.Sprintf("%s; ", command)
+
+		for envName := range envVar {
+			cmd += fmt.Sprintf("$env:%s=$null; ", envName)
```

Comment:
You should restore to its original value. It may be an issue if the user set `PATH` for example.

Reply:
Oh you're right. Nice catch!

Reply:
Should be fixed in [commit](https://github.com/DataDog/datadog-agent/pull/19798/commits/7e35202e6a62fb1afc283dd6526801a715940b20)

Reply:
Why should we restore env values ? Aren't we setting env vars only for the given command, prefixing it ? It should not affect the ssh session

Reply:
I see this is for Windows only !

Reply:
Yes on Windows ENV variables are set globally and not only for the command, so we need to restore them afterwards